pkg/snapshotter/btrfs.go

```diff
@@ -726,10 +709,53 @@ func (b *Btrfs) setBtrfsForFirstTime(rootDev string) error {
 	}
 
 	b.cfg.Logger.Debug("Create btrfs quota group")
-	cmdOut, err = b.cfg.Runner.Run("btrfs", "qgroup", "create", "1/0", b.rootDir)
+	cmdOut, err = b.cfg.Runner.Run("btrfs", "qgroup", "create", "1/0", state.MountPoint)
+	if err != nil {
+		b.cfg.Logger.Errorf("failed creating quota group for %s: %s", state.MountPoint, string(cmdOut))
+		return err
+	}
+	return b.remountStatePartition(state)
+}
+
+func (b *Btrfs) configureSnapperAndRootDir(state *v1.Partition) error {
+	rootDir, stateMount, err := findStateMount(b.cfg.Runner, state.Path)
 	if err != nil {
-		b.cfg.Logger.Errorf("failed creating quota group for %s: %s", b.rootDir, string(cmdOut))
+		b.cfg.Logger.Errorf("failed setting snapper root and state partition mountpoint: %v", err)
 		return err
 	}
-	return b.remountStatePartition(rootDev)
+
+	state.MountPoint = stateMount
+	b.rootDir = rootDir
+
+	if b.rootDir != "/" {
+		b.snapperArgs = []string{"--no-dbus", "--root", b.rootDir}
+	}
+	return nil
+}
+
+func findStateMount(runner v1.Runner, device string) (rootDir string, stateMount string, err error) {
```

Comment:
This is only executed on Active and Passive modes and essentially checks the current state mountpoints and sets the current snapshot root to the snapshotter root and the sets the state mountpoint to the actual top level volume (this is the `/run/initramfs/elemental-state` mountpoint).